### PR TITLE
Add cdrom test under throttle

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:1924604
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=throttle_cdrom_test --guestname=RHEL.8.3.0 --clone=no
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.throttle_cdrom_test: FAIL: Fail to get cdrom device with drv throttle after change (51.81 s)

Same result on windows guest.
Failed case due to production bug Bug 1922964 - [throttle] lost throttle attributes on cdrom device after execute blockdev-change-medium operation

**Comment by qingwangrh** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/2907

**Comment by qingwangrh** [comment]
@YongxueHong @zixi-chen Please help to review.

**Comment by zixi-chen** [comment]
Hi @qingwangrh ,

I search blockdev-change-medium on official website, this command combines blockdev-open-tray, blockdev-remove-medium, blockdev-insert-medium and blockdev-close-tray. 

I try not to use blockdev-open-tray, this is what I got, I think we can remove this cmd blockdev-open-tray:
**{"execute": "**blockdev-change-medium**", "arguments": {"id": "cdx", "filename": "/tmp/new.iso"}, "id": "ZZGpMwFM"}**
{"timestamp": {"seconds": 1612436084, "microseconds": 854034}, "event": "DEVICE_TRAY_MOVED", "data": {"device": "", "tray-open": true, "id": "cdx"}}
{"timestamp": {"seconds": 1612436084, "microseconds": 854396}, "event": "DEVICE_TRAY_MOVED", "data": {"device": "", "tray-open": false, "id": "cdx"}}

**{"execute": "query-block", "id": "8UyNu3zX"}**
{"return": [{"io-status": "ok", "device": "", "locked": false, "removable": false, "inserted": {"iops_rd": 0, "detect_zeroes": "off", "image": {"virtual-size": 21474836480, "filename": "/home/kvm_autotest_root/images/rhel840-64-virtio-scsi.qcow2", "cluster-size": 65536, "format": "qcow2", "actual-size": 5036253184, "format-specific": {"type": "qcow2", "data": {"compat": "1.1", "compression-type": "zlib", "lazy-refcounts": false, "refcount-bits": 16, "corrupt": false, "extended-l2": false}}, "dirty-flag": false}, "iops_wr": 0, "ro": false, "node-name": "drive_image1", "backing_file_depth": 0, "drv": "qcow2", "iops": 0, "bps_wr": 0, "write_threshold": 0, "encrypted": false, "bps": 0, "bps_rd": 0, "cache": {"no-flush": false, "direct": true, "writeback": true}, "file": "/home/kvm_autotest_root/images/rhel840-64-virtio-scsi.qcow2", "encryption_key_missing": false}, "qdev": "image1", "type": "unknown"}, {"io-status": "ok", "device": "", "locked": false, "removable": true, "inserted": {"iops_rd": 0, "detect_zeroes": "off", "image": {"virtual-size": 10842112, "filename": "/tmp/new.iso", "format": "raw", "actual-size": 10842112, "dirty-flag": false}, "iops_wr": 0, "ro": false, "node-name": "#block112", "backing_file_depth": 0, "drv": "raw", "iops": 0, "bps_wr": 0, "write_threshold": 0, "encrypted": false, "bps": 0, "bps_rd": 0, "cache": {"no-flush": false, "direct": false, "writeback": true}, "file": "/tmp/new.iso", "encryption_key_missing": false}, "qdev": "cdx", "tray_open": false, "type": "unknown"}], "id": "8UyNu3zX"}

**Comment by qingwangrh** [comment]
> Hi @qingwangrh ,
> 
> I search blockdev-change-medium on official website, this command combines blockdev-open-tray, blockdev-remove-medium, blockdev-insert-medium and blockdev-close-tray.
> 
> I try not to use blockdev-open-tray, this is what I got, I think we can remove this cmd blockdev-open-tray:
> **{"execute": "**blockdev-change-medium**", "arguments": {"id": "cdx", "filename": "/tmp/new.iso"}, "id": "ZZGpMwFM"}**
> {"timestamp": {"seconds": 1612436084, "microseconds": 854034}, "event": "DEVICE_TRAY_MOVED", "data": {"device": "", "tray-open": true, "id": "cdx"}}
> {"timestamp": {"seconds": 1612436084, "microseconds": 854396}, "event": "DEVICE_TRAY_MOVED", "data": {"device": "", "tray-open": false, "id": "cdx"}}
> 
> **{"execute": "query-block", "id": "8UyNu3zX"}**
> {"return": [{"io-status": "ok", "device": "", "locked": false, "removable": false, "inserted": {"iops_rd": 0, "detect_zeroes": "off", "image": {"virtual-size": 21474836480, "filename": "/home/kvm_autotest_root/images/rhel840-64-virtio-scsi.qcow2", "cluster-size": 65536, "format": "qcow2", "actual-size": 5036253184, "format-specific": {"type": "qcow2", "data": {"compat": "1.1", "compression-type": "zlib", "lazy-refcounts": false, "refcount-bits": 16, "corrupt": false, "extended-l2": false}}, "dirty-flag": false}, "iops_wr": 0, "ro": false, "node-name": "drive_image1", "backing_file_depth": 0, "drv": "qcow2", "iops": 0, "bps_wr": 0, "write_threshold": 0, "encrypted": false, "bps": 0, "bps_rd": 0, "cache": {"no-flush": false, "direct": true, "writeback": true}, "file": "/home/kvm_autotest_root/images/rhel840-64-virtio-scsi.qcow2", "encryption_key_missing": false}, "qdev": "image1", "type": "unknown"}, {"io-status": "ok", "device": "", "locked": false, "removable": true, "inserted": {"iops_rd": 0, "detect_zeroes": "off", "image": {"virtual-size": 10842112, "filename": "/tmp/new.iso", "format": "raw", "actual-size": 10842112, "dirty-flag": false}, "iops_wr": 0, "ro": false, "node-name": "#block112", "backing_file_depth": 0, "drv": "raw", "iops": 0, "bps_wr": 0, "write_threshold": 0, "encrypted": false, "bps": 0, "bps_rd": 0, "cache": {"no-flush": false, "direct": false, "writeback": true}, "file": "/tmp/new.iso", "encryption_key_missing": false}, "qdev": "cdx", "tray_open": false, "type": "unknown"}], "id": "8UyNu3zX"}

Due to blockdev-change-medium not always 100% succeed.  and it will not affect the info block result. 
More detail please refer to 1787009,1837185,1898032,1820506

**Comment by zixi-chen** [comment]
Checked bug, developer explained it makes sense guest boot with cdrom locked. Hence, add blockdev-open-tray with "force": true is reasonable.

LGTM,
(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_cdrom_test: FAIL: Fail to get cdrom device with drv throttle after change (71.64 s)

File: qemu/tests/cfg/throttle_cdrom_test.cfg [hunk 1]
```diff
@@ -0,0 +1,22 @@
+- throttle_cdrom_test:
+    type = throttle_cdrom_test
+    virt_test_type = qemu
+    kill_vm = yes
+    monitor_type = qmp
+    pre_command += "dd if=/dev/urandom of=/tmp/orig bs=10M count=1 && dd if=/dev/urandom of=/tmp/new bs=10M count=1 && mkisofs -o /tmp/orig.iso /tmp/orig && mkisofs -o /tmp/new.iso /tmp/new;"
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/tests/cfg/throttle_cdrom_test.cfg]: Updated,thanks.

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**